src/imu/imu.cpp

```diff
@@ -0,0 +1,242 @@
+    ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(iorientation->getOrientationSensorMeasureAsRollPitchYaw(0, rpyValues, timestamp), "Unable to obtain rpy measurements.");
+
+    iDynTree::Rotation I_R_FK = kinDynComp.getWorldTransform(frameName).getRotation(); 
+    I_R_I_IMU = (I_R_FK * ((iDynTree::Rotation::RPY(iDynTree::deg2rad(rpyValues[0]), iDynTree::deg2rad(rpyValues[1]), iDynTree::deg2rad(rpyValues[2]))).inverse())); 
+    
+    double minLim;
+    double maxLim;
+
+    for (int i = kinDynComp.model().getJointIndex("neck_pitch"); i <= kinDynComp.model().getJointIndex("neck_yaw"); i++)
+    {
+        ilim->getLimits(i, &minLim, &maxLim);
+
+        moveJoint(i, minLim + 5);
+        yarp::os::Time::delay(1.);
+        moveJoint(i, maxLim - 5);
+        yarp::os::Time::delay(1.);
+    }
+}
+
+bool Imu::sendData(iDynTree::Vector3 expectedValues, iDynTree::Vector3 imuSignal)
+{
+    yarp::os::Bottle& out = outputPort.prepare();
+    static yarp::os::Stamp stamp;
+
+    stamp.update();
+    outputPort.setEnvelope(stamp);
+    out.clear();
+
+    out.addFloat64(iDynTree::rad2deg(expectedValues[0]));
+    out.addFloat64(iDynTree::rad2deg(expectedValues[1]));
+    out.addFloat64(iDynTree::rad2deg(expectedValues[2]));
+
+    out.addFloat64(iDynTree::rad2deg(imuSignal[0]));
+    out.addFloat64(iDynTree::rad2deg(imuSignal[1]));
+    out.addFloat64(iDynTree::rad2deg(imuSignal[2]));
+
+    out.addFloat64(stamp.getTime());
+
+    outputPort.writeStrict();    
+    
+    return true;
+}
+
+bool Imu::moveJoint(int ax, double pos)
+{
+    bool done = false;
+    double refPos;
+
+    yarp::os::Time::delay(.1);
+
+    ipos->positionMove(ax, pos);
+    ipos->getTargetPosition(ax, &refPos);
+
+    while(!done)
+    {
+        iorientation->getOrientationSensorMeasureAsRollPitchYaw(0, rpyValues, timestamp);
+
+        ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(ienc->getEncoders(positions.data()), "Cannot get joint positions");
+        ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(ienc->getEncoderSpeeds(velocities.data()), "Cannot get joint positions");
+
+        for (auto i = 0; i < axes; i++)
+        {
+            s.setVal(i, iDynTree::deg2rad(positions[i]));
+            ds.setVal(i, iDynTree::deg2rad(velocities[i]));
+        }  
+
+        kinDynComp.setRobotState(
+        I_T_base,
+        s,
+        baseVelocity,
+        ds,
+        gravity);
+
+        iDynTree::Rotation expectedImuSignal = kinDynComp.getWorldTransform(frameName).getRotation();
+        iDynTree::Rotation imuSignal = (I_R_I_IMU * iDynTree::Rotation::RPY(iDynTree::deg2rad(rpyValues[0]), iDynTree::deg2rad(rpyValues[1]), iDynTree::deg2rad(rpyValues[2]))); 
+
+        auto error = (expectedImuSignal * imuSignal.inverse()).log();
+        double err_mean = (std::accumulate(error.begin(), error.end(), 0)) / error.size();
+        ROBOTTESTINGFRAMEWORK_ASSERT_ERROR_IF_FALSE(err_mean < 0.1, "Error > 0.1! Aborting ...");
```

Comment:
Similar to @Nicogene comments, in the future, the threshold could be a parameter, as if for some model/sensors are quite accurate, you may want to make sure that they contibute to be accurate.